### DISCUSSION TITLE
websocket with cookie or session

### DISCUSSION THREAD

**Discussion by FlaviuRadulescu** [walkor/Workerman#881] [Q&A]
Hi,

is there a way to implement some kind of session, cookie, unique identifier of a websocket connection/browser? I want to be able to identify websocket sesion behind a non sticky ha proxy so that I could route the request to any backend server?

If a browser reconnect after a temporary internet connection fault, I would like to be able to identify which was the browser session before the internet problem.

Thank you,
F

**Comment by walkor** [comment]
Workerman's websocket does not need session cookies. You can store data on the connection object, and these data can be obtained from the connection object at any time.

For example
```php
$worker = new Worker('websocket://0.0.0.0:1234');
$worker->onMessage = function($connection, $data)
{
    $connection->send($connection->lastData ?? 'none');
    $connection->lastData = $data;
};
```

**Reply by FlaviuRadulescu** [reply]
Walkor, thank you for the fast answer.

How does this work in case of a session stickiness of a HA proxy? How would the load balancer knows where to route the request?

Regarding your example, If I add some data to the connection object and somehow the request end up on another sever (going through the reconnect and everything) would my response (the one coming from the browser) contain this data that I have previously set?

Is there like an example of how you can scale and load balance the web sockets using a load balancer ? I am thinking to use the HA Proxy

Thanks,

**Comment by walkor** [comment]
I have no experience in using haproxy. I generally use nginx proxy to achieve load balancing. The configuration is similar to this.

```
server {
  
  upstream workerman {
      server 192.168.0.1:8282;
      server 192.168.0.2:8282;
  }

  location /
  {
    proxy_pass workerman;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header X-Real-IP $remote_addr;
  }

}
```
